docs/source/extensions/logging.rst

```diff
@@ -75,55 +108,79 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
-except functions with `batch_start` in their names.
+Use the :meth:`~pytorch_lightning.core.lightning.LightningModule.log`
+method to log from anywhere in a :doc:`lightning module <../common/lightning_module>` and :doc:`callbacks <../extensions/callbacks>`.
 
 .. code-block:: python
 
     def training_step(self, batch, batch_idx):
         self.log("my_metric", x)
 
 
-    # or a dict
+    # or a dict to get multiple metrics on the same plot of the logger supports it
     def training_step(self, batch, batch_idx):
         self.log("performance", {"acc": acc, "recall": recall})
 
-Depending on where log is called from, Lightning auto-determines the correct logging mode for you. \
-But of course you can override the default behavior by manually setting the :func:`~~pytorch_lightning.core.lightning.LightningModule.log` parameters.
+Depending on where log is called from, Lightning auto-determines the correct logging mode for you. But of course you can
+override the default behavior by manually setting the :meth:`~pytorch_lightning.core.lightning.LightningModule.log` parameters.
 
 .. code-block:: python
 
     def training_step(self, batch, batch_idx):
         self.log("my_loss", loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
 
-The :func:`~~pytorch_lightning.core.lightning.LightningModule.log` method has a few options:
-
-* `on_step`: Logs the metric at the current step. Defaults to `True` in :func:`~~pytorch_lightning.core.lightning.LightningModule.training_step`, and :func:`~pytorch_lightning.core.lightning.LightningModule.training_step_end`.
-
-* `on_epoch`: Automatically accumulates and logs at the end of the epoch. Defaults to True anywhere in validation or test loops, and in :func:`~~pytorch_lightning.core.lightning.LightningModule.training_epoch_end`.
-
-* `prog_bar`: Logs to the progress bar.
-
-* `logger`: Logs to the logger like Tensorboard, or any other custom logger passed to the :class:`~pytorch_lightning.trainer.trainer.Trainer`.
-
+The :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method has a few options:
+
+* ``on_step``: Logs the metric at the current step.
+* ``on_epoch``: Automatically accumulates and logs at the end of the epoch.
+* ``prog_bar``: Logs to the progress bar (Default: ``False``).
+* ``logger``: Logs to the logger like ``Tensorboard``, or any other custom logger passed to the
+    :class:`~pytorch_lightning.trainer.trainer.Trainer` (Default: ``True``).
+
+.. list-table:: Default behavior of logging in Callback or LightningModule
+   :widths: 50 25 25
+   :header-rows: 1
+
+   * - Hook
+     - on_step
+     - on_epoch
+   * - on_before_backward, on_after_backward, on_before_optimizer_step, on_before_zero_grad
+     - False
+     - True
+   * - on_train_start, on_train_epoch_start, on_train_epoch_end, training_epoch_end
+     - False
+     - True
+   * - on_train_batch_start, on_train_batch_end
```

Comment:
don't resolve this one.. need to update once https://github.com/PyTorchLightning/pytorch-lightning/pull/10756 gets merged.